### ISSUE TITLE
Support for tracking.

### ISSUE THREAD

**Issue by luboszk** [redis/go-redis#1247]
Hi,

I'm checking currently that library to use it in the project which I'm working on.
As part of that, I need to use client-side caching which uses tracking. Link to the Redis: [documentation](https://redis.io/topics/client-side-caching)

Can anyone tell me if it's possible to right now use that feature or there are still some parts missing? I was not able to find any documentation/code about tracking so I assume support for that does not exist at all.
If anyone could point me into the direction of how it should be implemented I can work on that.

Regards,
Lubosz.

**Comment by fighterlyt** [comment]
does anyone try to work on this ,or can i work on this?

**Comment by vmihailenco** [comment]
@fighterlyt AFAIK none is working on this. And I still did not get any experience with client side caching in Redis.

**Comment by bdollma** [comment]
Hi, 
Are there any plans to support this?
Thanks,

**Comment by gabzim** [comment]
I've been curious to implement this, I'm thinking of using the approach where you redirect all your invalidations to a pubsub channel: https://redis.io/docs/manual/client-side-caching/#two-connections-mode (I was thinking of using this approach instead of pushing the invalidation on the same connection to avoid breaking the requests/response model ingrained in the source code of the library)

The problem is that for every connection in the pool you need to execute the client tracking command and to redirect invalidations to specific client id. The `pool` module offers no "run this for every connection" and while I can run it for idle connections I am not sure how to handle some connection being busy etc without the logic becoming unmanageable. Then you also need to handle reconnects etc.

**Comment by minghsu0107** [comment]
@gabzim I was also working on how to integrate go-redis with Redis client side caching. However I also feel that it's quite awkward to manage connection IDs. My workaround is that I publish invalidation messages to a normal Redis pubsub channel, and subscribe to the channel in a separate groroutine.

**Comment by matyat** [comment]
I was curious about using this redis feature so I decided to have a go at creating a POC myself if anyone is interested. I used the approach mentioned above with a subscription and, as pointed out, getting the client ID of a `redis.PubSub` is a bit awkward.

Implemented for a `redis.Ring`, but I'm sure similar approaches could be used for a plain `redis.Client` or `redis.ClusterClient`.

https://gist.github.com/matyat/5738dcf323aa17a73ed333e73523198e
